src/amo/api/collections.js

```diff
@@ -285,3 +287,79 @@ export const createCollection = ({
     _validateLocalizedString,
   });
 };
+
+type ModifyCollectionAddonBaseParams = {|
+  addonId: number,
+  api: ApiStateType,
+  collectionSlug: string,
+  user: string,
+  _modifyCollectionAddon?: (any) => Promise<void>,
+|};
+
+export type CreateCollectionAddonParams = {|
+  ...ModifyCollectionAddonBaseParams,
+  notes?: CollectionAddonNotes,
+|};
+
+export type UpdateCollectionAddonParams = {|
+  ...ModifyCollectionAddonBaseParams,
+  notes: CollectionAddonNotes,
+|};
+
+type ModifyCollectionAddonParams =
+  | {| action: 'create', ...CreateCollectionAddonParams |}
+  | {| action: 'update', ...UpdateCollectionAddonParams |};
+
+export const modifyCollectionAddon = (
+  params: ModifyCollectionAddonParams,
+): Promise<void> => {
+  const { action, api, user, collectionSlug, addonId } = params;
```

Comment:
Should these be in alphabetical order, or do we not care about that in this case?

Reply:
Personally I don't really care (I use an editor with good name finding capabilities) but I can change it. I don't feel strongly about it. I'll adjust the others too.